project/.gitlab-ci.yml

```diff
@@ -59,3 +63,25 @@ python3.7:
 python3.8:
   <<: *test
   image: python:3.8
+
+python3.9:
+  <<: *test
+  image: python:3.9
+
+
+pages:
+  stage: deploy
+  image: python:3.6
+  before_script:
+    - pip install poetry
+    - poetry --version
+    - poetry install -vv
+  script:
+    - poetry run duty test
+    - poetry run duty coverage
+    - poetry run duty docs-deploy-gitlab
```

Comment:
This time, instead of re-running the tests, we should bring the coverage report from the test job, as an artifact. Not sure how to do this in GitHub Workflows, I will search again and report back.

Reply:
I think i have this all setup properly now in latest version let me know what you think.